jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/exception/QueryExceptionHandler.java

```diff
@@ -57,31 +78,65 @@ public static DataCloudJDBCException createException(String message, Exception e
                     try {
                         errorInfo = firstError.unpack(ErrorInfo.class);
                     } catch (InvalidProtocolBufferException exc) {
-                        return new DataCloudJDBCException("Invalid error info", e);
+                        return new DataCloudJDBCException("Invalid error info for query " + queryId, e);
                     }
 
-                    String sqlState = errorInfo.getSqlstate();
-                    String customerHint = errorInfo.getCustomerHint();
-                    String customerDetail = errorInfo.getCustomerDetail();
-                    String primaryMessage = String.format(
-                            "%s: %s%nDETAIL:%n%s%nHINT:%n%s",
-                            sqlState, errorInfo.getPrimaryMessage(), customerDetail, customerHint);
-                    return new DataCloudJDBCException(primaryMessage, sqlState, customerHint, customerDetail, ex);
+                    grpcMessage = status.getMessage();
+                    sqlState = errorInfo.getSqlstate();
+                    primaryMessage = errorInfo.getPrimaryMessage();
+                    customerHint = errorInfo.getCustomerHint();
+                    customerDetail = errorInfo.getCustomerDetail();
                 }
             }
         }
-        return new DataCloudJDBCException(message, e);
-    }
 
-    public static SQLException createException(String message, String sqlState, Exception e) {
-        return new SQLException(message, sqlState, e.getCause());
-    }
+        StringBuilder sharedMessageBuilder = new StringBuilder("Failed to execute query");
+        // We use status.getMessage() as in contrast to the primary message this already has the trace id
+        // encoded.
+        if (grpcMessage != null) {
+            sharedMessageBuilder.append(": " + grpcMessage);
+        } else {
+            // If we don't have any information from the server we resort to using the exception message
+            sharedMessageBuilder.append(": " + e.getMessage());
+        }
+        sharedMessageBuilder.append(String.format("%nSQLSTATE: %s", sqlState));
```

Comment:
use modern string interpolation instead of `String.format`

Here and other places